src/components/PopupWithForm.js

```diff
@@ -13,11 +14,16 @@ export default class PopupWithForm extends Popup {
 
   _setEventListeners() {
     this.#inputElements = this._popupForm.querySelectorAll(".modal__input");
+    this.#submitButton = this._popupForm.querySelector(".modal__button");
 
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this.close();
+      const originalText = this.#submitButton.textContent;
+      this.#submitButton.textContent = "Saving...";
+      this._handleFormSubmit(this._getInputValues()).then(() => {
+        this.#submitButton.textContent = originalText;
```

Comment:
`  this.#submitButton.textContent = originalText;`


You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.